tree/dataframe/test/dataframe_datasetspec.cxx

```diff
@@ -0,0 +1,229 @@
+{
+   auto dfWriter = RDataFrame(5).Define("x", [](ULong64_t e) { return int(e); }, {"rdfentry_"});
+   dfWriter.Snapshot<int>("tree", "file.root", {"x"});
+
+   // "reference" values
+   const auto dfSimple = RDataFrame("tree", "file.root").Display<int>({"x"})->AsString();
+   const auto dfRange0 = RDataFrame("tree", "file.root").Range(2, 4).Display<int>({"x"})->AsString();
+
+   // specify only tree and file names
+   const auto dfRDS0 = RDataFrame(RDatasetSpec("tree", "file.root")).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS0, dfSimple);
+
+   // specify tree name, file name, starting index => [2, 0) => range is disregarded (all entries are displayed)
+   const auto dfRDS1 = RDataFrame(RDatasetSpec("tree", "file.root", 2)).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS1, dfSimple);
+
+   // specify tree name, file name, starting index, ending index, where both indices are valid => [2, 4)
+   const auto dfRDS2 = RDataFrame(RDatasetSpec("tree", "file.root", 2, 4)).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS2, dfRange0);
+
+   // specify explicitly the column
+   const auto dfRDS3 = RDataFrame(RDatasetSpec("tree", "file.root", 2, 4, {"x"})).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS3, dfRange0);
+
+   // specify explicitly the column twice (and this time not explicitly requesting 1 column from Display), this produces
+   // 1 column
+   const auto dfRDS4 = RDataFrame(RDatasetSpec("tree", "file.root", 2, 4, {"x", "x"})).Display()->AsString();
+   EXPECT_EQ(dfRDS4, dfRange0);
+
+   // specify the treename as fifth argument => the first argument becomes the name of the chain of trees
+   const auto dfRDS5 =
+      RDataFrame(RDatasetSpec("chain", "file.root", 2, 4, {}, {"tree"})).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS5, dfRange0);
+
+   // specify the chain to have the same name as the tree
+   const auto dfRDS6 =
+      RDataFrame(RDatasetSpec("tree", "file.root", 2, 4, {}, {"tree"})).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS6, dfRange0);
+
+   // specify 2 trees, second tree is irrelevant, this is correct
+   const auto dfRDS7 =
+      RDataFrame(RDatasetSpec("chain", "file.root", 2, 4, {}, {"tree", "nottree"})).Display<int>({"x"})->AsString();
+   EXPECT_EQ(dfRDS7, dfRange0);
+
+   // specify 2 trees, first tree is irrelevant, this is wrong, emitting C++ error and ROOT error
+   EXPECT_THROW(
+      try {
+         ROOT_EXPECT_ERROR(RDataFrame(RDatasetSpec("chain", "file.root", 2, 4, {}, {"nottree", "tree"}))
+                              .Display<int>({"x"})
+                              ->AsString(),
+                           "TChain::LoadTree", "Cannot find tree with name nottree in file file.root");
+      } catch (const std::runtime_error &err) {
+         EXPECT_EQ(std::string(err.what()),
+                   "Column \"x\" is not in a dataset and is not a custom column been defined.");
+         throw;
+      },
+      std::runtime_error);
+
```

Comment:
Also this is not intended behaviour

Reply:
So, do we expect only 1 ROOT error, saying: `"<TChain::LoadTree> Cannot find tree with name nottree in file file.root"`?

Reply:
I don't know what the issue is with github. I commented the part where you construct the RDatasetSpec but then it is visualised much below, seemingly randomly. Disregard this comment, I will just merge and upload PR to upstream